compiler/src/iree/compiler/Codegen/Common/GPU/GPUDistributionPatterns.cpp

```diff
@@ -377,13 +378,17 @@ struct DistributeReductions final
                                 vector::CombiningKind combiningKind,
                                 int64_t entriesPerVector, Value mEmpty,
                                 OpBuilder &rewriter, Location loc) const {
-    uint32_t size = maxBitsPerShuffle;
+    auto funcOp = result.getDefiningOp()->getParentOfType<func::FuncOp>();
+    std::optional<int64_t> subgroupSize = getSubgroupSize(funcOp);
+    if (!subgroupSize)
+      funcOp->emitError("Could not find subgroup size!");
+
     Value mask;
     assert(llvm::isPowerOf2_64(laneSize));
     for (uint64_t i = shuffleOffset; i < shuffleOffset * laneSize; i <<= 1) {
       Value packed = packVectorToSupportedWidth(loc, rewriter, result);
-      auto shuffleOp = rewriter.create<gpu::ShuffleOp>(loc, packed, i, size,
-                                                       gpu::ShuffleMode::XOR);
+      auto shuffleOp = rewriter.create<gpu::ShuffleOp>(
+          loc, packed, i, subgroupSize.value(), gpu::ShuffleMode::XOR);
```

Comment:
Since the previous error does cause this to exit, I think this needs to be something like `.value_or(kDefaultSubgroupSize)`. Then we could also turn the error into a warning.

Reply:
32 seems like a safe default (suboptimal but supported on amdgpu, correct on nvidia)

Reply:
sure I can add that. thanks!